### ISSUE TITLE
Link Relations are not serialized

### ISSUE THREAD

**Issue by AD1337** [friflo/Friflo.Engine.ECS#33]
On `3.0.0-preview.15`
It seems that Link Relations are not currently serialized.

1. Is this by design, or are they expected to be serializable now or in an upcoming update?
2. Are all other [Relations ](https://friflo.gitbook.io/friflo.engine.ecs/examples/component-types#link-relation) also not serializable?

**Comment by AD1337** [comment]
Link **Components** are actually serializing and deserializing just fine.

So maybe I'm doing something wrong with Link **Relations**?

I tried:
`characterECS.AddRelation(new LinkCharacterToEndeavor(endeavor));`
And without a constructor:
`characterECS.AddRelation(new LinkCharacterToEndeavor { target = endeavor});`

Where the link struct is:
```
public struct LinkCharacterToEndeavor : ILinkRelation
{
    public Entity target;
    public Entity GetRelationKey() => target;
}
```

The Link Relation is created, it appears as a component in the debugger and I can use it fine:
![link](https://github.com/user-attachments/assets/ea699a35-6113-45c5-9dd5-34d4dab1321a)
But it's not serialized.

**Comment by AD1337** [comment]
Relation Components are also not serialized, just like Link Relations.

I could only serialize Link Components.

Am I doing something wrong, or is this the current state?

**Comment by friflo** [comment]
Serialization of relations is currently not supported.
Exactly this was a topic on Discord yesterday.
It is planned to implement this.

**Comment by friflo** [comment]
Just published `3.0.0-preview.17` on nuget which adds support for serialization of relations.
See https://friflo.gitbook.io/friflo.engine.ecs/examples/component-types#serialization

**Comment by friflo** [comment]
Support for serialization of relations as JSON is now available
See: https://friflo.gitbook.io/friflo.engine.ecs/package/release-notes#id-3.0.0-preview.17
